src/ol/source/wmtssource.js

```diff
@@ -465,12 +486,43 @@ ol.source.WMTS.optionsFromCapabilities = function(wmtsCap, config) {
   }
   goog.asserts.assert(urls.length > 0, 'At least one URL found');
 
+  var getFeatureInfoOptions = {};
+  if (wmtsCap['OperationsMetadata'].hasOwnProperty('GetFeatureInfo') &&
+          (config['featureInfoRequestEncoding'] !=
+          ol.source.WMTSRequestEncoding.REST)) {
+    getFeatureInfoOptions['requestEncoding'] =
+        ol.source.WMTSRequestEncoding.KVP;
+    gets =
+        wmtsCap['OperationsMetadata']['GetFeatureInfo']['DCP']['HTTP']['Get'];
+
+    constraint = goog.array.find(gets[0]['Constraint'],
```

Comment:
I wonder if we have a similar problem here as we had in: https://github.com/openlayers/ol3/pull/3409 with only checking the first entry?